### ISSUE TITLE
Add a reference section for containers.

### ISSUE THREAD

**Issue by rodrigogiraoserrao** [Textualize/textual#2032]
Other than a couple of random mentions in the Layout guide and the barebones API page, there isn't a great deal of info/examples for the containers.

A short reference might make sense, especially now that we have `Center`/`Middle` as of #2030.

**Comment by willmcgugan** [comment]
Agree. A section in https://textual.textualize.io/reference/ makes sense
